### ISSUE TITLE
/latest_changes and --page-file-dir

### ISSUE THREAD

**Issue by jgn** [gollum/gollum#1307]
Say you are serving with `--page-file-dir`

Can `/latest_changes` be constrained to only show changes within `--page-file-dir`? Maybe this should be automatic.

**Comment by bartkamphorst** [comment]
I think this would require changes in both gollum and gollum-lib. In gollum (`app.rb`), the `/latest_changes` route would have to pass the `page-file-dir path` in the options hash to the `latest_changes` call:
```ruby
# now
@wiki.latest_changes({:max_count => max_count})
# proposed fix
@wiki.latest_changes({:max_count => max_count, :path => @wiki.page_file_dir})
```

In gollum-lib (`wiki.rb`), the `latest_changes` method would have to pass the path along to `@repo.log`:
```ruby
# now
@repo.log(@ref, nil, options)
# proposed fix
@repo.log(@ref, options[:path], options)
```

I suspect that this would resolve this issue, but this solution is untested as of yet.

**Comment by bartkamphorst** [comment]
For this use case yes, but what about when we decide that `latest_changes` should show the latest changes for a specific page? 

This is something I find strange anyway, that hitting the Latest Changes button while on a specific page gives you a global overview of the wiki's commit history.

**Comment by dometto** [comment]
Yes, that would require an additional (optional) parameter to be passed to `gollum-lib`. That's a good idea, but the page file dir thing should not be a parameter I think, since that's a property of the wiki object.

**Comment by bartkamphorst** [comment]
I don't know if this changes how you feel about the fix, but the page_file_path is not passed as a parameter but as an option in the options hash (notice the curly brackets).

**Comment by dometto** [comment]
Right, but I guess the thing I find weird is to pass information in a public method call to an object (the `Wiki`) when that information is already passed along when the object is initialized.

**Comment by dometto** [comment]
No objections. However, there is already the `History` button for a page's "local" changes. Shouldn't there be a "global" Latest Changes, too?

**Comment by bartkamphorst** [comment]
Ah, you are completely right about the `History` button! This is off-topic from the current issue, but I would then suggest to just not show the `Latest Changes` button while on a specific page, only when you are on the landing page (Home). Would that make sense to you @dometto ?

**Comment by bartkamphorst** [comment]
This issue has now been resolved in the `gollum-lib-5.x` branch.

**Comment by dometto** [comment]
> I would then suggest to just not show the Latest Changes button while on a specific page, only when you are on the landing page (Home). Would that make sense to you @dometto ?

Yes, that would be one option. But perhaps there could also be a way to keep access to the Latest Changes from each page, but  somehow have it be visually clear that it's a "global" wiki option? Maybe it would make sense to mark the difference between page-specific and global buttons in general. We could defer the decision until we tackle the layout for `5.x`

> This issue has now been resolved in the gollum-lib-5.x branch.

👍 Closing, then. @jgn the fix will be available in the `5.0` release (which may be a while, though). Thanks for reporting.

**Comment by jgn** [comment]
Sweet!
